gms/gossiper.cc

```diff
@@ -2086,10 +2086,18 @@ future<> gossiper::add_saved_endpoint(inet_address ep) {
     auto ep_state = endpoint_state();
     auto es = get_endpoint_state_ptr(ep);
     if (es) {
+        if (es->get_heart_beat_state().get_generation()) {
+            logger.warn("Attempted to add saved endpoint after endpoint_state was already established with gossip: {}, at {}", es->get_heart_beat_state(), current_backtrace());
```

Comment:
May be worthwhile to log "ep" so that we know for which endpoint we have a collision.

Reply:
Oops, definitely worth printing it.